### PR TITLE
Added default repository

### LINKED ISSUE #10: A default repository
What do you think of configuring a default repository: One that will be used when the repository name parameter is omitted.

At the moment, anything using the ResourceBundle needs to know the repository name. As this name is purely based on the runtime configuration, every bundle needing something from the ResourceBundle is required to have a configuration option for this option.

In the case of the treeBrowserBundle, this would mean that anyone using the tree form type needs to know the repository name.

I think it would be much easier to have a default repository (are there many use-cases for multiple repositories?).

This could be done by either having a convention that the default repository is called `default`:
```yaml
cmf_resource:
    repositories:
        default:
            # ...
```

Or a new option can be added:
```yaml
cmf_resource:
    default_repository_name: phpcrodm
    repositories:
        phpcrodm:
            # ...
```

/cc @dantleech

### PR DESCRIPTION
**Pull request by WouterJ**
Fixes #10

### CODE REVIEW

File: Tests/Unit/Registry/ContainerRepositoryRegistryTest.php [hunk 1]
```diff
@@ -20,35 +20,59 @@ class ContainerRepositoryRegistryTest extends ProphecyTestCase
...
         $this->registry = new ContainerRepositoryRegistry($this->container->reveal(), $serviceMap);
     }
 
     public function testGet()
     {
-        $res = $this->registry->get('one');
-        $this->assertSame($res, $this->repo1->reveal());
+        $this->container->get('repo1_service')->shouldBeCalled();
+
+        $this->registry->get('one');
```
Comment (dantleech) [inline Tests/Unit/Registry/ContainerRepositoryRegistryTest.php]: Why change this? Now we are not checking that the repository is returned.

File: Tests/Unit/Registry/ContainerRepositoryRegistryTest.php [hunk 2]
```diff
@@ -20,35 +20,59 @@ class ContainerRepositoryRegistryTest extends ProphecyTestCase
...
+        if (false !== $defaultId) {
+            $registry = new ContainerRepositoryRegistry($this->container->reveal(), array(), array(), 'the_default');
+        } else {
+            $registry = new ContainerRepositoryRegistry($this->container->reveal());
+        }
+
+        $registry->get();
+    }
+
+    public function getGetDefaultRepositoryData()
```
Comment (dantleech) [inline Tests/Unit/Registry/ContainerRepositoryRegistryTest.php]: I always put the provider before the test, and call it `provideGetDefaultRepository`

File: Tests/Unit/Registry/ContainerRepositoryRegistryTest.php [hunk 3]
```diff
@@ -20,35 +20,59 @@ class ContainerRepositoryRegistryTest extends ProphecyTestCase
     public function setUp()
     {
         $this->container = $this->prophesize('Symfony\Component\DependencyInjection\ContainerInterface');
-        $this->repo1 = $this->prophesize('Puli\Repository\Api\ResourceRepository');
-        $this->repo2 = $this->prophesize('Puli\Repository\Api\ResourceRepository');
```
Comment (dantleech) [inline Tests/Unit/Registry/ContainerRepositoryRegistryTest.php]: I put any reusable mocks in the `setUp` by convention.

**Comment by WouterJ** [comment]
Changed the tests again.

**Comment by WouterJ** [comment]
And now there are also passing tests, not sure what went wrong yesterday (some vendor library).

**Comment by dantleech** [comment]
Cheers @WouterJ

**Comment by dantleech** [comment]
hmm. master has failed..

**Comment by WouterJ** [comment]
Yeah, it's a really weird issue. For some reason, it installs doctrine dbal 2.4 instead of 2.5, causing the installation of lower versions of the phpcr/jackalope libraries. This means the `--force` option that's used to bootstrap the database does not exists and building the database fails.

I've spent some hours debugging this yesterday and today, it suddently passed...

**Pull request merged**